src/App.jsx

```diff
@@ -14,45 +20,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+let isSorted = false;
+
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState('');
+
+  const sortByAbt = () => {
+    setVisibleGoods(
+      [...visibleGoods].sort(),
+      setSortField(SORT_BY_ABT),
+      isSorted = true,
+    );
+  };
+
+  const sortByLength = () => {
+    setVisibleGoods(
+      [...visibleGoods].sort((good1, good2) => good1.length - good2.length),
+      setSortField(SORT_BY_LENGTH),
+      isSorted = true,
+    );
+  };
+
+  const reverse = () => {
+    setVisibleGoods(
+      [...visibleGoods].reverse(),
+      setSortField(SORT_BY_REVERS),
+      isSorted = true,
+    );
+  };
```

Comment:
Good start, but it looks quite complicated and there are a lot of duplications in the code. Instead, we can create a function that will accept 3 parameters, products, a filtering method, and an isReversed parameter, and based on these parameters, the function should return a sorted array